platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,860 @@
+3. **Bilinear Pairings API**: Layer of abstraction to be included at compilation time providing the cryptography primitives
+   and the arithmetic operations for working with a specific EC curve and the underlying Groups, Fields, and Pairings
+   and minimizes the impact of changing or including different curve implementations.
+4. **Bilinear Pairings Impl**: This is an implementation of the Bilinear Pairings API that will be loaded at runtime using Java’s SPI.
+   Multiple implementations can be provided to support different types of curves. It allows changing or switching dependencies.
+5. **Native Support Lib**: Provides a set of generic functions loading native libraries in different system architectures when packaged in a jar
+   using a predefined organization so they can be accessed with JNI.
+6. **Arkworks[https://github.com/arkworks-rs]**:  arkworks is a Rust ecosystem for zkSNARK programming.
+   Our implementation uses it as the library responsible for elliptic curve cryptography.
+7. **EC-Key Utils ** is a utility module that enables the node operator to generate a bootstrapping public/private key pair.
+
+### Module organization and repositories
+![img_6.png](img_6.png)
+1. **hedera-cryptography**: This is a separate repository for hosting cryptography-related libraries.
+   It is necessary to facilitate our build process, which includes Rust libraries. It also provides independent release cycles between consensus node code and block node code.
+2. **swirlds-native-support**: Gradle module that enables loading into memory compiled native libraries so they can be used with JNI.
+3. **swirlds-cryptography-tss**: Gradle module for the TSS Library.
+   This library's only client is the consensus node, so it will be in the hedera-services repository, under the `platform-sdk` folder.
+4. **swirlds-cryptography-signatures**: Gradle module for the Bilinear Pairings Signature Library.
+5. **swirlds-cryptography-pairings-api**: Gradle module for the Bilinear Pairings API. Minimizes the impact of adding or removing implementations.
+6. **swirlds-cryptography-alt128**: Gradle module that will implement the Bilinear Pairings API using alt-128 elliptic curve.
+   That curve has been chosen due to EVM support. The artworks rust library will provide the underlying cryptography implementation.
+   The module will include Java and Rust codes that will be compiled for all possible system architectures and distributed in a jar under a predefined structure.
+
+### Libraries Specifications
+
+#### Swirlds Crypto TSS Library
+##### Overview
+This library implements the Groth21 TSS-specific primitives.
+
+Two options are presented, while the first focus on usability from the client perspective, the second one proposes operations that are more atomic focusing on providing clients with the ability of concurrently handle the library's operation.
```

Comment:
> while the first focus on usability from the client perspective

Nothing wrong with describing different options, but IMO option 1 is _not_ more usable from the client's perspective. The client is the platform, and these shared data structures make the library more difficult to use by the platform.

Reply:
@alittley ended up merging both proposals into one.
As I said, there are no mutable shared data structures.

Reply:
This seems reasonable.